src/DataModel/IMetadataProvider.ts

```diff
@@ -0,0 +1,30 @@
+import {IMetadataDescription} from "./IMetadataDescription";
+import "whatwg-fetch";
+import {IData} from "./IData";
+
+/**
+ * @interface Describes an abstract meta-data providing facility which translates from raw @link Response to an abstract data model.
+ */
+export interface IMetadataProvider
```

Comment:
I wanted to introduce an intermediate layer that would isolate the client and it's hypermedia controls internal data model from actual RDF serialization received. This would be a job for a *IMetadataProvider* to make that translation. In case of JSON-LD it would be i.e. framing.
As for the *IData* and *IHypermedia* - it is an attempt to isolate raw data that the client expected from the hypermedia controls so these can be easily used in the code.